src/formatDate.js

```diff
@@ -7,8 +7,54 @@
  *
  * @returns {string}
  */
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const parts = date.split(fromFormat[fromFormat.length - 1]);
+
+  const dateParts = {
+    DD: parts[fromFormat.indexOf('DD')],
+    MM: parts[fromFormat.indexOf('MM')],
+    YYYY: parts[fromFormat.indexOf('YYYY')],
+    YY: parts[fromFormat.indexOf('YY')],
```

Comment:
Using `indexOf` for 'YY' and 'YYYY' assumes both are present in `fromFormat`, which may not be the case. Consider adding checks to ensure these formats exist before accessing them.